### PR TITLE
ticket-2 complete

### PR DESCRIPTION
**Pull request by TheRealImrano**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+const express = require("express");
+const { getTopics } = require("./controllers/topicsController");
+
+
+const app = express();
+app.use(express.json());
```
Comment (nc-poppy-m) [inline app.js]: We're trying to write only the minimum amount of code needed to meet the requirements of the ticket, so I would take app.use(express.json()) out until you need it :)

File: app.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+app.get('/api/topics', getTopics)
+
+app.use((err, req, res, next) => {
```
Comment (nc-poppy-m) [inline app.js:9]: You aren't using this PSQL error handler for this, so it shouldn't be added until you get to a ticket where you need it

File: __tests__/app.test.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+const app = require('../app.js');
+
+beforeEach(() => {
+    return seed(data);
+});
+afterAll(() => {
+    return db.end();
+});
+
+test("returns 404 bad request when sent invalid endpoint", () => {
```
Comment (nc-poppy-m) [inline __tests__/app.test.js]: As you're not sending back an error from the models using a ```Promise.reject```, I think the 404s you're receiving are likely to the default ones coming from express, and not from the nice custom error handler you made in ```app.js```. This is completely fine for this ticket - but we don't have to test for express default behaviours, so either change your models so the error is passed to your custom error handler, or just let express deal with it and delete the test and custom error handler until you need it. 🙂 

To check which one it is, try sending a different message back to your custom error handler and testing that.

File: app.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    if (err.code === '22P02') {
+      res.status(400).send({
+        msg: "bad request - invalid SQL",
+        code: err.code
+      })
+    }
+    next(err)
+})
+
+app.use((err, req, res, next) => {
```
Comment (nc-poppy-m) [inline app.js]: If you run your 404 test - do you get this ```console.log``` ?

**Review by nc-poppy-m — CHANGES_REQUESTED** [review]
Hi Ali! Really good work on this ticket, just a few considerations that I've outlined in here (mostly about your error handling). Get these changed and send another pr in once you're done 😄

**Comment by TheRealImrano** [comment]
Thank you for the feedback Poppy! 

Yes, it seems the custom handlers weren't in use just yet, so i removed them. Ill resubmit nchelp pr once i push up the new changes

**Review by TheRealImrano — COMMENTED** [review]
Ive made edits to my toclet 2. are they visible?

**Review by nc-poppy-m — APPROVED** [review]
All good to merge!

**Comment by TheRealImrano** [comment]
thanks!

**Pull request merged**